lib/extractor/layer.ts

```diff
@@ -135,6 +201,48 @@ function getLayersContentAndArchiveManifest(
   };
 }
 
+function getOciLayersContentAndArchiveManifest(
+  imageIndex: OciImageIndex,
+  manifestCollection: OciArchiveManifest[],
+  layers: Record<string, ExtractedLayers>,
+): { layers: ExtractedLayers[]; manifest: OciArchiveManifest } {
+  // filter empty layers
+  // get the layers content without the name
+  // reverse layers order from last to first
+
+  // get manifest file first
+  const manifest = getOciManifest(imageIndex, manifestCollection);
+  const filteredLayers = manifest.layers
+    .filter((layer) => Object.keys(layers[layer.digest]).length !== 0)
+    .map((layer) => layers[layer.digest])
+    .reverse();
+
+  return {
+    layers: filteredLayers,
+    manifest,
+  };
+}
+
+function getOciManifest(
+  imageIndex: OciImageIndex,
+  manifestCollection: OciArchiveManifest[],
+): OciArchiveManifest {
+  if (imageIndex.manifests.length === 1) {
+    return manifestCollection[0];
+  }
+
+  const manifest = imageIndex.manifests.find(
+    (item) =>
+      item.platform?.architecture === "amd64" && item.platform.os === "linux",
```

Comment:
why theres `?` on the left part and not also in the right part? is it intentional?

Reply:
The `&&` expression is evaluated left to right, so if platform is `null` or `undefined`, it's returned as `false` and if it's not `false` it means that platform is not `null`, so we don't need the `?`, I hope I'm right?! 😆

Reply:
yes :) great explanation! I would still put it in both just for the case someone will change the order of the condition or something like that, but really not critical